src/main/resources/com/google/api/codegen/java/settings.snip

```diff
@@ -105,7 +115,12 @@
      * Returns the default service endpoint.
      */
     public static String getDefaultEndpoint() {
-      return "{@xsettingsClass.serviceAddress}:{@xsettingsClass.servicePort}";
+      @switch xsettingsClass.transportProtocol
+      @case "GRPC"
+        return "{@xsettingsClass.serviceAddress}:{@xsettingsClass.servicePort}";
+      @case "HTTP"
+        return "{@xsettingsClass.serviceAddress}";
```

Comment:
Why no port for http?

Reply:
The discovery doc schema didn't have a field for port, so I didn't know what to put as a port. Calls to a URL without a port 80 or 443 explicitly specified work just fine. I assume that the Http transport classes from google-http-client-java would take care of that?

Reply:
https = 443, so hardcode it somewhere.

Reply:
Done.